pyneuroml/neuron/__init__.py

```diff
@@ -126,3 +135,558 @@ def export_to_neuroml1(hoc_file, nml1_file_name, level=1, validate=True):
+            'avg_rev_pot': 'NA',
+            'int_conc': 'NA',
+            'ext_conc': 'NA',
+        }
+
+    # https://neuronsimulator.github.io/nrn/python/modelspec/programmatic/mechtype.html#MechanismType
+    mt = h.MechanismType(0)
+    mname = h.ref('')
+    pname = h.ref('')
+
+    for i in range(mt.count()):
+        mt.select(i)
+        mt.selected(mname)
+        # provide storage for mechanism
+        # https://neuronsimulator.github.io/nrn/python/programming/mechstan.html#MechanismStandard
+        ms = h.MechanismStandard(mname, 1)
+        numParams = ms.count()
+
+        # construct empty dicts to hold information
+        totParamVal = {}
+        paramsectiondict = {}  # type: dict[typing.Any, typing.Any]
+        for j in range(numParams):
+            # assign pname the name of the jth parameter
+            ms.name(pname, j)
+            totParamVal[j] = 0
+            paramsectiondict[pname[0]] = {}
+
+        numSecPresent = 0
+        for sec in seclist:
+            if (h.ismembrane(mname, sec=sec)):
+                numSecPresent += 1
+                # segment information is provided as a fraction of the total
+                # section length, not the segment list
+                seginfo = {}
+                for seg in range(1, sec.nseg + 1):
+                    mid_pt = get_seg_midpoint(seg, sec.nseg)
+                    logger.debug(f"section {sec}: {seg}/{sec.nseg}: {mid_pt}")
+                    ms._in(mid_pt, sec=sec)
+                    for j in range(numParams):
+                        ms.name(pname, j)
+                        totParamVal[j] += ms.get(pname)
+                        if not (str(sec)) in paramsectiondict[pname[0]].keys():
+                            paramsectiondict[pname[0]][str(sec)] = {}
+                        seginfo[seg] = ms.get(pname)
+
+                values = seginfo.values()
+                unique_values = list(set(values))
+                # if all values are the same, only print them once as '*'
+                for j in range(numParams):
+                    ms.name(pname, j)
+                    if len(unique_values) == 1:
+                        paramsectiondict[pname[0]][str(sec)] = {
+                            "nseg": sec.nseg,
+                            "values": {
+                                '*': unique_values[0]
+                            }
+                        }
+                    else:
+                        paramsectiondict[pname[0]][str(sec)] = {
+                            "nseg": sec.nseg,
+                            "values": seginfo
+                        }
+
+        mt_dict = {
+            'present_on_secs': numSecPresent,
+            'num_params': numParams,
+            'parameters': {
+            }
+        }
+
+        for j in range(numParams):
+            ms.name(pname, j)
+            try:
+                param_dict = {
+                    'ave_all_sections': totParamVal[j] / numSecPresent,
```

Comment:
No, if we pass two sections as args to the function, then it will be an issue. If we are restricting the fucntion to only a section then we can go ahead with ave_all_segs. 

And if we plan to go with passing multiple sections as args then we should output `ave_all_sections` and [changing this line](https://github.com/NeuroML/pyNeuroML/blob/e019841058e38c59ad4388cc0afd4d72ef6213e5/pyneuroml/neuron/__init__.py#L455) with `numSecPresent += sec.nseg + 1` will solve it?

Reply:
Yeh, but not exactly. When we use the segments to get values, we're taking values from the centre of each segment to add up. When we take it only from the section (and not each segment), it only gives the value of the first segment:

https://neuronsimulator.github.io/nrn/python/programming/mechstan.html?highlight=mechanismstandard#MechanismStandard._in

So this average metric isn't a good one in general. If we do sections, then what it's actually saying is "average over all first segments (of all sections)", which isn't good for non-homogenous mechs either. 

A better way of calculating this would be over all segments for each section, and reporting that value for each section individually. But, would that be any more useful than the weak metric we have now?

Reply:
All this average metric is doing is helping us cross-check the parameter's homogeneity, other than that I see no use for calculating it on all sections level. The way `ave_all_sections` is calculated won't make sense at all. It's better to skip it and instead report for each section individually.